Feature/num 395 integration tests invalid profile body temperature

af-pwo opened the pull request.

wlad commented on pom.xml:
```diff
@@ -26,7 +26,7 @@
         <spring-boot.version>2.3.4.RELEASE</spring-boot.version>
         <woodstox.version>6.1.1</woodstox.version>
-        <ehrbase-sdk.version>v0.3.7</ehrbase-sdk.version>
+        <ehrbase-sdk.version>0.3.7</ehrbase-sdk.version>
```
I guess v0.3.7 is to outdated for @SevKohler(?)

We should consider creating and pushing a new version of openEHR_SDK.

OR

as stated in https://jitpack.io/docs/ we could  use `develop-SNAPSHOT` as dependency version. Jitpack would then build openEHR_SDK from latest commit on develop branch.

OR

we could pin the version to some commit hash from develop branch like
```
<version>33aa416</version>
```

ppazos replied:
@wlad other branches don't have build errors and have that version, please check https://github.com/ehrbase/fhir-bridge/blob/feature/iss10_bodyHeight2/pom.xml#L29

So I think 'v0.3.7' it's OK, but '0.3.7' is not. Though I'm not against using develop-SNAPSHOT.

I think we decided to avoid using specific commit hashes as dependencies in the POMs, so I would say 'no' to option 3 :P

SevKohler replied:
This 

> 
> 
> > @ppazos hope now i fixed everything so far, can you please check again, thanks.
> > We still need to discuss changing the build number @ppazos @wlad.
> > In think i prefer the solution with the SNAPSHOT, commit numbers need to be changed manually and just using only a newer sdk version will force us to update it too once in a while. With the Snapshot we can always jump back to an older version (if bugs appear) while still using the newest one.
> 
> @SevKohler can you check committing with 'v0.3.7' to see if the CI passes? Other branches using that version build OK.

Already tested this several times, here you go. Please remember the blood gas mapping will not work with this version!

SevKohler replied:
I reference you and explained the situation in this [issue](https://github.com/ehrbase/fhir-bridge/issues/84) if you remember. (0.3.7 is the build number of my local openEHR_SDK version (newer one), this is the one we need to ¨replace¨).

Unfortunatly our current v0.3.7 is not ¨OK¨ since it includes several bugs that have been fixed, the builds and unit tests work because they do not test the actual content of the Composition that was created, they merely check if there was no error as i [explained ](https://github.com/ehrbase/fhir-bridge/issues/66) here.

SevKohler reviewed (commented):
@ppazos hope now i fixed everything so far, can you please check again, thanks.

We still need to discuss changing the build number @ppazos @wlad.
In think i prefer the solution with the SNAPSHOT, commit numbers need to be changed manually and just using only a newer sdk version will force us to update it too once in a while. With the Snapshot we can always jump back to an older version while still using the newest one.

ppazos commented:
> @ppazos hope now i fixed everything so far, can you please check again, thanks.
> 
> We still need to discuss changing the build number @ppazos @wlad.
> In think i prefer the solution with the SNAPSHOT, commit numbers need to be changed manually and just using only a newer sdk version will force us to update it too once in a while. With the Snapshot we can always jump back to an older version (if bugs appear) while still using the newest one.

@SevKohler can you check committing with 'v0.3.7' to see if the CI passes? Other branches using that version build OK.

SevKohler commented:
> 
> 
> > @ppazos hope now i fixed everything so far, can you please check again, thanks.
> > We still need to discuss changing the build number @ppazos @wlad.
> > In think i prefer the solution with the SNAPSHOT, commit numbers need to be changed manually and just using only a newer sdk version will force us to update it too once in a while. With the Snapshot we can always jump back to an older version (if bugs appear) while still using the newest one.
> 
> @SevKohler can you check committing with 'v0.3.7' to see if the CI passes? Other branches using that version build OK.

Tested this already several times, here you go. Please remember, blood gas mapping will not work with this version!

ppazos commented:
> Sonarcloud is failing, because the [new test class](https://github.com/ehrbase/fhir-bridge/blob/feature/60_mapping_blood_gas/src/test/java/org/ehrbase/fhirbridge/BundleTest.java) (as far as i can see it) is not taken into the coverage.

@SevKohler can you move the new tests the the current IT file? I think that is the only thing needed to merge, to make the sonar pass.

SevKohler commented:
> 
> 
> > Sonarcloud is failing, because the [new test class](https://github.com/ehrbase/fhir-bridge/blob/feature/60_mapping_blood_gas/src/test/java/org/ehrbase/fhirbridge/BundleTest.java) (as far as i can see it) is not taken into the coverage.
> 
> @SevKohler can you move the new tests the the current IT file? I think that is the only thing needed to merge, to make the sonar pass.

Locally i have over 50% coverage. I disagree, tests for Bundle should be contained in a seperate class. Putting them all in one class is bad coding, i wont do that. Please take the time and configure the sonarcloud properly, because for the next mappings there i will be also new classes.

ppazos commented:
@SevKohler what I suggested was just to try it and see if sonar didn't complain about it, I don't have strong opinions on where to put test. Though I don't think we have a guide on good or bad test organization, any references to good practices for test organization are welcome :)

@wlad can you check why Severin's other test file doesn't seem to be taken into account by the sonar analysis?

About the test failing because of the version, did you try the SNAPSHOP version suggested by Wlad?

SevKohler commented:
> @SevKohler what I suggested was just to try it and see if sonar didn't complain about it, I don't have strong opinions on where to put test. Though I don't think we have a guide on good or bad test organization, any references to good practices for test organization are welcome :)

Do a class for every mapping type at least, or better a package, as it is practice in OOP. Since we should write more tests anyway test classes for each mapping of a type should be contained in this packages. 
tests/Bundle/BloodGasMappingTest 
tests/Observation/BloodpressureTest
tests/Observation/BodyWeightTest

SevKohler commented:
> will do so tomorrow!
develop-SNAPSHOT works !
Problematic is now that i get this [bug](https://github.com/ehrbase/openEHR_SDK/issues/120) which appears randomly, now again and wont change even if rebuilding. @ppazos may you be so nice and build the project and see if this bug appears also for you ? If yes we need to tell stefan to fix this.

wlad commented:
> @wlad can you check why Severin's other test file doesn't seem to be taken into account by the sonar analysis?

yes, it was ignored by maven failsafe plugin because it's name was not following the pattern `*IT.java`. I rename testfile to fix that.
But now two test from BundleIT.java are failing. Please review them @SevKohler  @ppazos

ppazos commented:
Thanks @wlad I guess @SevKohler needs to check your changes and double check all tests are working and being executed by the CI.

Current errors here seem to be because of data not validating with the OPT rules https://app.circleci.com/pipelines/github/ehrbase/fhir-bridge/413/workflows/a5f53aa2-1625-4c87-8db2-6f5e3e7d4904/jobs/615

SevKohler commented:
@wlad your the man, nice !
@ppazos this error is currently solved by rebuilding the ehrbase and appears spontanously, i already [reported ](https://github.com/ehrbase/openEHR_SDK/issues/120) it to stefan. I thought its only related to my build, that why i wanted to wait for the CI. Know its sure.

SevKohler commented:
Blocked by ehrbase/openEHR_SDK/[#120](https://github.com/ehrbase/openEHR_SDK/issues/120)

SevKohler commented:
@wlad @ppazos the bug was fixed in the SDK ehrbase/openEHR_SDK/#120, as far as i can see it the bug appears again in the [CI](https://app.circleci.com/pipelines/github/ehrbase/fhir-bridge/493/workflows/df7e9de3-167d-404b-a619-a459d5717290/jobs/786/tests), can one of you confirm this is not a problem of the CI config (i guess). So i may reopen it.

SevKohler commented:
@wlad @ppazos the bug was fixed in the SDK ehrbase/openEHR_SDK/[#122](https://github.com/ehrbase/openEHR_SDK/pull/122) as far as i can see it the bug appears again in the [CI](https://app.circleci.com/pipelines/github/ehrbase/fhir-bridge/493/workflows/df7e9de3-167d-404b-a619-a459d5717290/jobs/786/tests), can one of you confirm this is not a problem of the CI config (old version of ehrbase), so i may reopen it.

related ehrbase/openEHR_SDK[#120](https://github.com/ehrbase/openEHR_SDK/issues/120)

af-pwo committed:
[ADD] observation-example-body-temperature-robot.json for testing purposes
[FIX] typo in heart rate test case
[ADD] testcases for invalid profile for body temperature

af-pwo committed:
[DEBUG] Debuged invalid test cases

af-pwo committed:
Merge branch 'develop' into feature/NUM-395_integrationTests_invalidProfile_bodyTemperature

# Conflicts:
#	tests/robot/OBSERVATION/04_create_heart_rate.robot

Commit:
renaming robot updated number from 5 to 6

af-dpe reviewed (approved):
code looks fine
